### ISSUE TITLE
Crash in DetoxSync

### ISSUE THREAD

**Issue by denissb** [wix/Detox#2641]
I have an issue when running tests in parallel with `--workers 2` using `jest-circus` after upgrading to detox 18
The form tests that use the keyboard actions like `typeText` and `clearText` fail when they are running in parallel with another test.

#### Reproduction

1. Have 2 tests, one that is verifying some on screen behaviour (fx. navigation or button actions) and one that is testing some form actions like entering and removing text in inputs to check form validation logic
2. Run the tests in parallel with `--workers 2` using `jest-circus` as a runner
3. See the tests fail with random errors and a sync issue

I also get this error thrown in one of the tests when running in parallel
```
thrown: "Signal 11 was raised
    (
    	0   Detox                               0x000000010aa97ef5 +[NSThread(DetoxUtils) dtx_demangledCallStackSymbols] + 37
    	1   Detox                               0x000000010aa9b100 __DTXHandleCrash + 464
    	2   Detox                               0x000000010aa9b845 __DTXHandleSignal + 59
    	3   libsystem_platform.dylib            0x00007fff6115a5fd _sigtramp + 29
    	4   ???                                 0x0000000000000000 0x0 + 0
    	5   DetoxSync                           0x000000010b861aaf -[_DTXTimerTrampoline fire:] + 188
    	6   DetoxSync                           0x000000010b849ef0 _DTXCFTimerTrampoline + 74
    	7   CoreFoundation                      0x00007fff20390c57 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
    	8   CoreFoundation                      0x00007fff2039072a __CFRunLoopDoTimer + 926
    	9   CoreFoundation                      0x00007fff2038fcdd __CFRunLoopDoTimers + 265
    	10  CoreFoundation                      0x00007fff2038a35e __CFRunLoopRun + 1949
    	11  CoreFoundation                      0x00007fff203896d6 CFRunLoopRunSpecific + 567
    	12  App                          0x00000001080f2fd1 +[RCTCxxBridge runRunLoop] + 993
    	13  DetoxSync                           0x000000010b8523c0 swz_runRunLoopThread + 291
    	14  Foundation                          0x00007fff20858e68 __NSThread__start__ + 1042
    	15  libsystem_pthread.dylib             0x00007fff61167109 _pthread_start + 148
    	16  libsystem_pthread.dylib             0x00007fff61162b8b thread_start + 15
    )"
```

#### Expected behaviour
The tests should pass just like when you run them with `--maxWorkers 1`

#### Environment (please complete the following information):
 - Detox: 18.3.1
 - React Native: 0.63.4
 - Node: 14.15.4
 - Device: iPhone 8
 - Xcode: 12.3
 - iOS: 14.4
 - macOS: 10.15.7

- [x] I have followed the instructions under [Identifying which synchronization mechanism causes us to wait too much](https://github.com/wix/Detox/blob/master/docs/Troubleshooting.Synchronization.md#identifying-which-synchronization-mechanism-causes-us-to-wait-too-much), I have read [synchronization debug documentation](https://github.com/wix/DetoxSync/blob/master/IdleStatusDocumentation.md) and am providing the relevant synchronization debug output below:

**Comment by LeoNatan** [comment]
This looks like a crash in DetoxSync. @alon-ha @d4vidi 

```
    	4   ???                                 0x0000000000000000 0x0 + 0
    	5   DetoxSync                           0x000000010b861aaf -[_DTXTimerTrampoline fire:] + 188
    	6   DetoxSync                           0x000000010b849ef0 _DTXCFTimerTrampoline + 74
```
It could be a bad app code, but it could be a bug in DetoxSync.

**Comment by LeoNatan** [comment]
I don't think the number of workers matters here.

**Comment by LeoNatan** [comment]
Might be similar to #2627 (which was closed in error as duplicate of another issue).

**Comment by denissb** [comment]
> I don't think the number of workers matters here.

Maybe it doesn't but when I run the tests without `--workers 2` they are rock solid and pass constantly. I also didn't face any issues with the same test suit running on Detox 17.14.8

**Comment by LeoNatan** [comment]
Detox 18 has a new synchronization mechanism, built in-house.

**Comment by denissb** [comment]
Yes, but it is quite nice. Thats why I want to upgrade, so now I am also digging around trying to find the cause.
I will try to upgrade to 18.6.0 and see if it changes anything.

**Comment by sam-barker** [comment]
@denissb Please comment if you find the same crash when upgrading to the latest v18. When I last tried it happened on 18.3

**Comment by d4vidi** [comment]
@alon-ha any leads to help out?

**Comment by denissb** [comment]
@sam-barker I still have the same issue on 18.6.0

**Comment by sam-barker** [comment]
Any movement on this at all?

**Comment by DanielMarkiel** [comment]
Hi,
I'm facing the same issue on my project but with `--maxWorkers 1`.
All tests are passing fine on Android emulator.
Tests were passing fine on iOS with Detox: 17.14.9.

*Environment:*
- Detox: 18.6.2
- jest-circus: 26.6.3
- React Native: 0.63.4
- Node: 14.16.0
- Device: iPhone 8
- Xcode: 12.4
- iOS: 14.4
- macOS: 11.2.1


Logs:
```
thrown: "Signal 11 was raised
    (
    	0   Detox                               0x000000010fec6eb5 +[NSThread(DetoxUtils) dtx_demangledCallStackSymbols] + 37
    	1   Detox                               0x000000010feca0c0 __DTXHandleCrash + 464
    	2   Detox                               0x000000010feca805 __DTXHandleSignal + 59
    	3   libsystem_platform.dylib            0x00007fff6115cd7d _sigtramp + 29
    	4   libsystem_c.dylib                   0x00007fff86d4a728 __global_locale + 0
    	5   CoreFoundation                      0x00007fff20490049 -[__NSDictionaryM dealloc] + 128
    	6   libobjc.A.dylib                     0x00007fff2018f834 objc_object::sidetable_release(bool, bool) + 174
    	7   libobjc.A.dylib                     0x00007fff2018bb85 _object_remove_assocations + 580
    	8   libobjc.A.dylib                     0x00007fff20188f25 objc_destructInstance + 84
    	9   libobjc.A.dylib                     0x00007fff2018efb8 -[NSObject dealloc] + 21
    	10  App                       0x000000010eac59a5 -[FBSDKGraphRequestConnection dealloc] + 96
    	11  libobjc.A.dylib                     0x00007fff2018f834 objc_object::sidetable_release(bool, bool) + 174
    	12  CFNetwork                           0x00007fff236d0b10 _CFNetworkHTTPConnectionCacheSetLimit + 163652
    	13  CFNetwork                           0x00007fff23519ec5 CFNetwork + 28357
    	14  DetoxSync                           0x0000000111d1b270 ____detox_sync_dispatch_wrapper_block_invoke + 23
    	15  libdispatch.dylib                   0x00007fff2010532f _dispatch_call_block_and_release + 12
    	16  libdispatch.dylib                   0x00007fff20106508 _dispatch_client_callout + 8
    	17  libdispatch.dylib                   0x00007fff2010c3f9 _dispatch_lane_serial_drain + 715
    	18  libdispatch.dylib                   0x00007fff2010cfab _dispatch_lane_invoke + 458
    	19  libdispatch.dylib                   0x00007fff20117577 _dispatch_workloop_worker_thread + 782
    	20  libsystem_pthread.dylib             0x00007fff61167499 _pthread_wqthread + 314
    	21  libsystem_pthread.dylib             0x00007fff61166467 start_wqthread + 15
    )"

   4 |
      5 | describe('Intro', () => {
    > 6 |   beforeAll(async () => {
        |   ^
      7 |     await device.launchApp(deviceLaunchAppConfig);
      8 |     await device.setURLBlacklist(blacklistedURLs);
      9 |   });
```

**Comment by d4vidi** [comment]
All, we are currently on a break and a bit short-staffed at the moment, in general. We hope to get on top of this within 1-2 weeks. Here are similar discussions that could be relevant, in the meantime:
* https://github.com/wix/Detox/issues?q=label%3A%22z_domain%3A+detox18%22+crash
* https://github.com/wix/DetoxSync/issues/5

**Comment by shamilovtim** [comment]
Hello, any updates on this? We are sorely in need of a Firebase upgrade (the old version of Firebase is actually causing bugs in our app) but we can't bump Firebase until this gets resolved.

**Comment by wdospinal** [comment]
Hi, an advice on this, the app is having the issue when trying to add firebase and segment to the project

**Comment by noomorph** [comment]
@alon-ha, pinging you just for visibility.

Personally, I think this is an important issue since many RN users are using Firebase including our teams as well if I am not mistaken.

Let's check if we already have clear instructions on how to reproduce the crash. @wdospinal, @shamilovtim could you help to locate any demo project crashing with that error or create one?

I think Alon will get back to his job duties starting from the next week, and I hope he'll find time during April-May to check out a demo project if it will be already prepared.

**Comment by shamilovtim** [comment]
From what I could tell based on shallow research the issue might be caused by monkey patching in HTTP libraries. That's supposedly how it appeared in KTOR: https://github.com/firebase/firebase-ios-sdk/issues/7277

I've seen other tickets in the ecosystem also blaming the same thing in various ways

**Comment by shamilovtim** [comment]
Also related: https://github.com/JetBrains/kotlin-native/pull/4569 (blame was again monkey patching)

**Comment by noomorph** [comment]
@shamilovtim, what but isn't that Android-related there?

**Comment by shamilovtim** [comment]
I believe it was kotlin-native (compiles to iOS) so that's why it was iOS relevant. The underlying issue caused a bug in iOS code but it connected to kotlin code.

**Comment by noomorph** [comment]
@shamilovtim ah, I never realized that Firebase was using there kotlin-native, okay, cool. 👍

**Comment by shamilovtim** [comment]
Please see here: #270

There is a track record of this kind of swizzling / reflection causing issues and breaking Firebase. I assume the fix is probably pretty similar

**Comment by sam-barker** [comment]
Is there any update on this? I assume a fair few projects are a major version behind now and still on v17 because of this issue. I know we certainly are

**Comment by shamilovtim** [comment]
We've having to work around this in a bad way. Other packages that were depending on Firebase were falling behind to the point of becoming show stoppers. Firebase is not optional in our project, multiple other packages depend on Firebase and Detox is not working with Firebase.

**Comment by shamilovtim** [comment]
More logs:

Crash 1:
```
2021-05-17 13:50:49.509 I  Shipt[30146:a0e86] [com.apple.network:connection] nw_endpoint_handler_cancel [C27 localhost:32301 ready resolver (satisfied (Path is satisfied), interface: en0)]
2021-05-17 13:50:49.509 I  Shipt[30146:a0e86] [com.apple.network:connection] nw_association_schedule_deactivation <nw_association localhost:32301> will become dormant after 10000ms of inactivity
2021-05-17 13:50:49.509 I  Shipt[30146:a0e86] [com.apple.network:connection] nw_endpoint_handler_cancel [C27.1 127.0.0.1:32301 ready socket-flow (satisfied (Path is satisfied), interface: lo0)]
2021-05-17 13:50:49.509 I  Shipt[30146:a0e86] [com.apple.network:connection] nw_association_schedule_deactivation <nw_association 127.0.0.1:32301> will become dormant after 10000ms of inactivity
2021-05-17 13:50:49.509 Df Shipt[30146:a0e86] [com.apple.network:connection] nw_flow_disconnected [C27.1 127.0.0.1:32301 cancelled socket-flow ((null))] Output protocol disconnected
2021-05-17 13:50:49.509 Db Shipt[30146:a0e86] [com.apple.network:connection] nw_resolver_cancel [C27] 0x7ffa0a538060
2021-05-17 13:50:49.509 Df Shipt[30146:a0e86] [com.apple.network:connection] nw_connection_report_state_with_handler_on_nw_queue [C27] reporting state cancelled
2021-05-17 13:50:49.510 I  Shipt[30146:a0e86] [com.apple.network:connection] [C27 localhost:32301 tcp, url hash: 08d3ccea] dealloc
2021-05-17 13:50:49.510 Df Shipt[30146:a0e86] (CFNetwork) Connection 27: done
2021-05-17 13:50:49.513 Db Shipt[30146:a0e86] [com.apple.CFNetwork:Coalescing] removing all entries config 0x60000052f020
2021-05-17 13:50:49.513 Df Shipt[30146:a0e86] (CFNetwork) Connection 33: cleaning up
2021-05-17 13:50:49.513 Df Shipt[30146:a0e86] [com.apple.network:connection] [C33 37907897-6175-411B-BE0F-58F6A5AEAC16 localhost:8081 tcp, url hash: f6991239] cancel
2021-05-17 13:50:49.513 Df Shipt[30146:a0e86] [com.apple.network:connection] [C33 37907897-6175-411B-BE0F-58F6A5AEAC16 localhost:8081 tcp, url hash: f6991239] cancelled
	[C33.1 4C68FBA4-F704-42B5-96F4-6C31EF693AE2 127.0.0.1:65467<->127.0.0.1:8081]
	Connected Path: satisfied (Path is satisfied), viable, interface: lo0
	Duration: 0.249s, DNS @0.001s took 0.003s, TCP @0.006s took 0.001s
	bytes in/out: 16650029/247, packets in/out: 1090/1, rtt: 0.000s, retransmitted bytes: 0, out-of-order bytes: 0
2021-05-17 13:50:49.513 I  Shipt[30146:a0e86] [com.apple.network:connection] nw_endpoint_handler_cancel [C33 localhost:8081 ready resolver (satisfied (Path is satisfied), interface: en0)]
2021-05-17 13:50:49.513 I  Shipt[30146:a0e86] [com.apple.network:connection] nw_association_schedule_deactivation <nw_association localhost:8081> will become dormant after 10000ms of inactivity
2021-05-17 13:50:49.513 I  Shipt[30146:a0e86] [com.apple.network:connection] nw_endpoint_handler_cancel [C33.1 127.0.0.1:8081 ready socket-flow (satisfied (Path is satisfied), interface: lo0)]
2021-05-17 13:50:49.513 I  Shipt[30146:a0e86] [com.apple.network:connection] nw_association_schedule_deactivation <nw_association 127.0.0.1:8081> will become dormant after 10000ms of inactivity
2021-05-17 13:50:49.513 Df Shipt[30146:a0e86] [com.apple.network:connection] nw_flow_disconnected [C33.1 127.0.0.1:8081 cancelled socket-flow ((null))] Output protocol disconnected
2021-05-17 13:50:49.513 Db Shipt[30146:a0e86] [com.apple.network:connection] nw_resolver_cancel [C33] 0x7ff9ea507820
2021-05-17 13:50:49.514 Df Shipt[30146:a0e86] [com.apple.network:connection] nw_connection_report_state_with_handler_on_nw_queue [C33] reporting state cancelled
2021-05-17 13:50:49.514 E  Shipt[30146:a0e87] [com.wix.Detox:DetoxManager] App crashed: Signal 11 was raised
(
	0   Detox                               0x0000000111515ef5 +[NSThread(DetoxUtils) dtx_demangledCallStackSymbols] + 37
	1   Detox                               0x0000000111519100 __DTXHandleCrash + 464
	2   Detox                               0x0000000111519845 __DTXHandleSignal + 59
	3   libsystem_platform.dylib            0x00007fff60335d7d _sigtramp + 29
	4   ???                                 0x0000700002bc6580 0x0 + 123145348212096
	5   CoreFoundation                      0x00007fff20492833 -[__NSDictionaryM dealloc] + 128
	6   libobjc.A.dylib                     0x00007fff20191604 objc_object::sidetable_release(bool, bool) + 174
	7   libobjc.A.dylib                     0x00007fff2018d8ad _object_remove_assocations + 562
	8   libobjc.A.dylib                     0x00007fff2018ac02 objc_destructInstance + 84
	9   libobjc.A.dylib                     0x00007fff20190d88 -[NSObject dealloc] + 21
	10  libobjc.A.dylib                     0x00007fff20191604 obj<…>
2021-05-17 13:50:49.514 I  Shipt[30146:a0e87] [com.wix.Detox:DetoxInstrumentsManager] Called stop but no recording in progress
2021-05-17 13:50:49.514 I  Shipt[30146:a0e86] [com.apple.network:connection] [C33 localhost:8081 tcp, url hash: f6991239] dealloc
2021-05-17 13:50:49.522 I  Shipt[30146:a0d10] [com.wix.Detox:WebSocket] Action received: reactNativeReload
```

Crash two:
```
2021-05-17 14:10:42.271 Df Shipt[30786:a7ebb] [com.apple.network:connection] nw_flow_disconnected [C29.1 172.217.4.42:443 cancelled socket-flow ((null))] Output protocol disconnected
2021-05-17 14:10:42.271 I  Shipt[30786:a7eca] [com.apple.runningboard:assertion] Acquiring assertion: <RBSAssertionDescriptor| "Shared Background Assertion 3 for com.shipt.groceries-staging" ID:(null) target:30786>
2021-05-17 14:10:42.271 Db Shipt[30786:a7ebb] [com.apple.network:connection] nw_resolver_cancel [C29] 0x7fb046c2ff60
2021-05-17 14:10:42.271 Df Shipt[30786:a7ebb] [com.apple.network:connection] nw_connection_report_state_with_handler_on_nw_queue [C29] reporting state cancelled
2021-05-17 14:10:42.271 Db Shipt[30786:a7ebb] [com.apple.CFNetwork:Coalescing] removing all entries config 0x600001ee6080
2021-05-17 14:10:42.271 Db Shipt[30786:a7ebb] [com.apple.network:boringssl] boringssl_bio_destroy(196)
2021-05-17 14:10:42.272 Db Shipt[30786:a7eca] [com.apple.runningboard:assertion] Adding assertion 24911-30786-1889 to dictionary
2021-05-17 14:10:42.272 I  Shipt[30786:a7ebb] [com.apple.network:connection] [C29 firebaselogging-pa.googleapis.com:443 tcp, url hash: cdff6ee4, tls] dealloc
2021-05-17 14:10:42.272 I  Shipt[30786:a7ebb] [com.apple.network:] nw_context_purge_endpoints Context <nw_context com.apple.CFNetwork.NSURLSession.{18F6C993-251C-4F4C-86F8-4BD2F4532047}{(null)}{Y}{2}> has more than 0 cache entries, purging 2 from 2 down to 0
2021-05-17 14:10:42.277 E  Shipt[30786:a7f28] [com.wix.Detox:DetoxManager] App crashed: Signal 11 was raised
(
	0   Detox                               0x0000000110710ef5 +[NSThread(DetoxUtils) dtx_demangledCallStackSymbols] + 37
	1   Detox                               0x0000000110714100 __DTXHandleCrash + 464
	2   Detox                               0x0000000110714845 __DTXHandleSignal + 59
	3   libsystem_platform.dylib            0x00007fff60335d7d _sigtramp + 29
	4   ???                                 0x000070000b3c1500 0x0 + 123145490797824
	5   CoreFoundation                      0x00007fff20492833 -[__NSDictionaryM dealloc] + 128
	6   libobjc.A.dylib                     0x00007fff20191604 objc_object::sidetable_release(bool, bool) + 174
	7   libobjc.A.dylib                     0x00007fff2018d8ad _object_remove_assocations + 562
	8   libobjc.A.dylib                     0x00007fff2018ac02 objc_destructInstance + 84
	9   libobjc.A.dylib                     0x00007fff20190d88 -[NSObject dealloc] + 21
	10  Foundation                          0x00007fff207f7ca1 -[N<…>
2021-05-17 14:10:42.277 I  Shipt[30786:a7f28] [com.wix.Detox:DetoxInstrumentsManager] Called stop but no recording in progress
2021-05-17 14:10:42.299 A  Shipt[30786:a7ebb] (Security) SecItemDelete_ios
2021-05-17 14:10:42.300 A  Shipt[30786:a7ebb] (Security) SecItemDelete_ios
2021-05-17 14:10:42.301 A  Shipt[30786:a7ebb] (Security) SecItemDelete_ios
2021-05-17 14:10:42.487 I  Shipt[30786:a7e81] [com.apple.UIKit:Application] Deactivation reason added: 12; deactivation reasons: 0 -> 4096; animating application lifecycle event: 0
```

**Comment by shamilovtim** [comment]
I have isolated the issue to `UIApplicationDelegate` swizzling. This bug is probably related to both detox and firebase swizzling `UIApplicationDelegate`

**Comment by d4vidi** [comment]
@alon-ha this could probably give us a good lead as to where the root of the problem is.

**Comment by alon-ha** [comment]
@shamilovtim Sorry for the delay. Can you please open an example project with both firebase and detox so I will try to solve it?

**Comment by shamilovtim** [comment]
If you have a default starter template you could point me to where you have built out some e2e tests I think it would just be a matter of adding Firebase to it, which I could do no problem. Then that could live on for debugging any future Firebase issues if/when they arise

**Comment by alon-ha** [comment]
We currently don't have a starter template. I'm working on one exactly for such cases

**Comment by ball-hayden** [comment]
Cross-posting here from https://github.com/wix/DetoxSync/issues/5#issuecomment-865851840

I've put together a minimal reproduction with Firebase Perf enabled: https://github.com/PlayerData/DetoxFirebaseRepro
Worth noting that produces a different trace to that originally reported in this issue - I think there might be multiple issues here that have been combined together.

**Comment by alon-ha** [comment]
@ball-hayden That's great, I will try to look on the example you created.
I created a general template which will help speed bugs like this which we need to reproduce locally.
So next time use [DetoxTemplate](https://github.com/wix-incubator/DetoxTemplate) :)

**Comment by alon-ha** [comment]
@ball-hayden actually in the readme I see that you wrote that we need to setup a firebase project and some plist file..
That's exactly the part I want to skip on and just start to debug it from a point which firebase already setup and the tests failed because of that.. Will be great if you can fork the [DetoxTemplate](https://github.com/wix-incubator/DetoxTemplate) and do so (just follow the readme there).

**Comment by ball-hayden** [comment]
@alon-ha I'm really sorry - I can't share a private key to do that.

**Comment by ball-hayden** [comment]
I don't think that would be possible @alon-ha. The missing plist file contains only API keys, and is downloaded from the Firebase Console. I've just tried populating that with fake details and the app fails to start - I assume the Firebase library is validating on startup.

I can fork DetoxTemplate, but after I have added React Native Firebase you would still need to create a Firebase "project" and download API keys.

If you aren't able to carry out steps 1-3 of https://firebase.google.com/docs/ios/setup (which I understand is a pain, but is only clicking through), I can privately send you the plist file for my test Firebase project?

**Comment by alon-ha** [comment]
> I don't think that would be possible @alon-ha. The missing plist file contains only API keys, and is downloaded from the Firebase Console. I've just tried populating that with fake details and the app fails to start - I assume the Firebase library is validating on startup.
> 
> I can fork DetoxTemplate, but after I have added React Native Firebase **you would still need to create a Firebase "project" and download API keys.**
> 
> If you aren't able to carry out steps 1-3 of https://firebase.google.com/docs/ios/setup (which I understand is a pain, but is only clicking through), I can privately send you the plist file for my test Firebase project?

I meant that you can open a new account in firebase and set all those things into a fork of DetoxTemplate.
Basically so the fork repo will already include everything.

**Comment by ball-hayden** [comment]
Unfortunately, I can't do that as that would still be sharing API keys publically (albeit for a test firebase account).

I'll get a fork ready. Either someone else who is willing to publish keys can do so, or we can exchange keys privately.

**Comment by shamilovtim** [comment]
They don't have accounts for OSS? Seems like something like that should exist

**Comment by ball-hayden** [comment]
@alon-ha my apologies - I've done a bit more reading into the `GoogleService-Info.plist` file and the contents are unique but considered to be non-secret.

I've pushed one to my original demo project.
I'll also get a fork of the DetoxTemplate fully setup.

**Comment by ball-hayden** [comment]
@shamilovtim you'd need a free account to reproduce this - all of the necessary bits are in the free "spark" plan.

**Comment by ball-hayden** [comment]
https://github.com/PlayerData/DetoxFirebaseReproTemplate
^ There's a fork with Firebase setup.

The issue is intermittent - I've adapted your provided test to run multiple times. I reliably hit the bug within the first few iterations.

**Comment by alon-ha** [comment]
@ball-hayden Thank you for the example. I'm hoping to start working on that this week.
I already checked the swizzling in detox and seems that it was done correctly. Hopefully I can fix it in our side, but my best solution might be to open a PR to Firebase. Will keep you guys update

**Comment by alon-ha** [comment]
@ball-hayden I debugged the Template you provided.
Unfortunately you can't  put the describe inside a for [loop](https://github.com/PlayerData/DetoxFirebaseReproTemplate/commit/be37eb32fe009a1f9c0811f838294d2ebd7f6bd0#diff-75c74b7610d9ef3a5650f1645bce501764378aac1602e6f98fa56da13e4cb879R1), that's not a valid way to run e2e tests - the for loop won't wait for each to finish..
This will fail without Firebase as well (you can try yourself).
Please write a test/senario which without Firebase [configure](https://github.com/PlayerData/DetoxFirebaseReproTemplate/commit/be37eb32fe009a1f9c0811f838294d2ebd7f6bd0#diff-c819eafab8ec2b7665c03cf159518493bc8e9521a66e40102b6d8283f40e4671R49) line it will pass and fail with Firebase.
Waiting for you to fix/add test which will demonstrate the actual problem

**Comment by ball-hayden** [comment]
Interesting. Is it possible that this bug is aggravated by Firebase, but does also exist without?

I've got to tap out for a bit, but I'll happily accept a PR to https://github.com/PlayerData/DetoxFirebaseReproTemplate from anyone else following this thread.

Otherwise, I'll update that repo when I've got some time available again.

I assume that simply duplicating the test file (instead of running in a loop) isn't enough to trigger the crash?

**Comment by alon-ha** [comment]
> Is it possible that this bug is aggravated by Firebase, but does also exist without
Everything is possible (: But running a describe block inside a loop won't work regardless..
So I need a valid example which reproduce the bug. 
As you assumed simply duplicating the test file won't trigger a crash.
The example should be something that without `[FIRApp configure];` will probably work right but crash with it

**Comment by shamilovtim** [comment]
If you have enough tests in the project you'll be more guaranteed to hit the bug from what I can tell

**Comment by alon-ha** [comment]
> If you have enough tests in the project you'll be more guaranteed to hit the bug from what I can tell

I disagree. Inside the repo we have a test project and each time detox version build on master, 33 tests files are running with an average of 3 test for each file.. So it's clearly not an issue of `enough tests`.
That's exactly why I want to see a template with firebase which I can reproduce and debug it (clearly an issue with some swizzling), but I need to see it locally in order to help here

**Comment by ball-hayden** [comment]
I've pushed a change to https://github.com/PlayerData/DetoxFirebaseReproTemplate which uses multiple files rather than a for loop, and it looks like duplicating the test file _does_ trigger the crash (admittedly, you have to duplicate it a few times...)

I've run the following:

```
npm run build:ios-release
npm run test:ios-release -- --workers 2
```

I'd didn't get a summary line from the output (for some reason), but I observe 5 fail with a Signal 11.

Sample Signal 11

```
 FAIL  e2e/ExampleScreen41.e2e.js (34.525 s)
  Example Screen Tests
    ✕ should type into the text field (5055 ms)

  ● Example Screen Tests › should type into the text field

    DetoxRuntimeError: The pending request #3 ("invoke") has been rejected due to the following error:

    The app has crashed, see the details below:

    Signal 11 was raised
    (
    	0   Detox                               0x00000001040e0c45 +[NSThread(DetoxUtils) dtx_demangledCallStackSymbols] + 37
    	1   Detox                               0x00000001040e3e50 __DTXHandleCrash + 464
    	2   Detox                               0x00000001040e4595 __DTXHandleSignal + 59
    	3   libsystem_platform.dylib            0x00007fff60335d7d _sigtramp + 29
    	4   ???                                 0xc766036a6e348a5b 0x0 + -4078568656887051685
    	5   CoreFoundation                      0x00000001070d5833 -[__NSDictionaryM dealloc] + 128
    	6   libobjc.A.dylib                     0x0000000106f07604 objc_object::sidetable_release(bool, bool) + 174
    	7   libobjc.A.dylib                     0x0000000106f038ad _object_remove_assocations + 562
    	8   libobjc.A.dylib                     0x0000000106f00c02 objc_destructInstance + 84
    	9   libobjc.A.dylib                     0x0000000106f06d88 -[NSObject dealloc] + 21
    	10  Foundation                          0x0000000104dcdca1 -[NSOperation dealloc] + 281
    	11  libobjc.A.dylib                     0x0000000106f07604 objc_object::sidetable_release(bool, bool) + 174
    	12  libsystem_blocks.dylib              0x0000000109cb5432 _Block_release + 128
    	13  libsystem_blocks.dylib              0x0000000109cb5432 _Block_release + 128
    	14  libsystem_blocks.dylib              0x0000000109cb5432 _Block_release + 128
    	15  DetoxTemplate                       0x000000010385dae6 __destroy_helper_block_e8_32s40s48s + 24
    	16  libsystem_blocks.dylib              0x0000000109cb5432 _Block_release + 128
    	17  libsystem_blocks.dylib              0x0000000109cb5432 _Block_release + 128
    	18  libdispatch.dylib                   0x0000000109b7c8df _dispatch_client_callout + 8
    	19  libdispatch.dylib                   0x0000000109b82f5c _dispatch_lane_serial_drain + 1042
    	20  libdispatch.dylib                   0x0000000109b8398c _dispatch_lane_invoke + 400
    	21  libdispatch.dylib                   0x0000000109b8df81 _dispatch_workloop_worker_thread + 772
    	22  libsystem_pthread.dylib             0x00007fff6034045d _pthread_wqthread + 314
    	23  libsystem_pthread.dylib             0x00007fff6033f42f start_wqthread + 15
    )
```

**Comment by ball-hayden** [comment]
> Unfortunately you can't put the describe inside a for loop, that's not a valid way to run e2e tests - the for loop won't wait for each to finish..

I feel it might be helpful to pull this thread a bit. This was previously written as a synchronous for loop, not an async `forEach` block - why would the loop not wait for each to finish?

**Comment by alon-ha** [comment]
Have some tasks but hopefully I will get to it in this week

**Comment by alon-ha** [comment]
From just looking over the stack trace I can say it seems like an issue which is not related to detox. 
I will say again, detox will raise Signal 11 whenever the app crash. When you see:
    	`1   Detox                               0x00000001040e3e50 __DTXHandleCrash + 464`
It's mean that detox catch the crash but it's not necessary the cause.
It's defiantly not seems to be related to the issue with firebase.

Again: "The example should be something that without [FIRApp configure]; will probably work right but crash with it"

@ball-hayden I will check what you provided as well, but I doubt it's the issue with firebase

**Comment by shamilovtim** [comment]
Disabling Firebase to be on while Detox runs fixed the issue for us, so there's an underlying incompatibility between Detox and Firebase. I don't think anyone made any judgement as to which side was responsible for this, only that the incompatibility exists.

**Comment by alon-ha** [comment]
@shamilovtim Are you talking about disabling Firebase in the provided [example](https://github.com/PlayerData/DetoxFirebaseReproTemplate) by @ball-hayden or in general?
I'm aware there is an incompatibility, I'm just trying to work/debug on example which show it

**Comment by ball-hayden** [comment]
@alon-ha I've just run https://github.com/PlayerData/DetoxFirebaseReproTemplate with `[FIRApp configure]` commented out, and all the tests pass.

**Comment by alon-ha** [comment]
Cool so maybe it is related. Will check soon

**Comment by dlitmano** [comment]
@ball-hayden that helped, thank you!

**Comment by alon-ha** [comment]
I pull the recent changes and did a complete clean install, both npm and pods.

Build in both configurations:
`npm run build:ios-release
npm run build:ios-debug`

Tried two times each of the following combinations:
`npm run test:ios-debug
npm run test:ios-debug -- --workers 2
npm run test:ios-release -- --workers 2
npm run test:ios-release -- --workers 4`

Each time 50/50 tests pass

**Comment by ball-hayden** [comment]
> @ball-hayden are the tests failing for you consistency?

The suite always has failures with Firebase enabled, yes. But each individual test, no - it seems to be an intermittent fault. For me it's less than 50/50 - it's about 1 in 10 that fail.

With `[FIRApp configure]` commented (and the app rebuilt!) the suite passes with no workers flag, `--workers 2`. With `--workers 4`, I get other failures but no Signal 11 (I think my machine can't quite handle 4 simulators at once!).

**Comment by sergiomap** [comment]
in our case if we remove `[FIRApp configure]` it breaks the app. 
Is there any fix planned regarding this problem with Firebase?

**Comment by SandroMachado** [comment]
@alon-ha do you have any updates?

**Comment by SandroMachado** [comment]
>@alon-ha I've just run https://github.com/PlayerData/DetoxFirebaseReproTemplate with [FIRApp configure] commented out, and all the tests pass.

This not a solution or a workaround because all the code that uses the firebase instance will crash.

**Comment by d4vidi** [comment]
@alon-ha let's pair on this, later this week 🙏🏻

**Comment by alon-ha** [comment]
@SandroMachado I returned from a vacation and will try to work on this soon. For me locally, I couldn't reproduce the bug, all the tests were passed each time with the provided template @ball-hayden wrote.
My guess is that we can find a scenario with a test which fail it each time with Firebase but pass without it. Then it will be easier to debug. I'll try to create such scenario and update you guys

**Comment by dirkpostma** [comment]
Similar issue here every now and then.

```
detox[52662] ERROR: [WS_ERROR] The app has crashed, see the details below:

Signal 11 was raised
(
	0   Detox                               0x0000000108fa1c45 +[NSThread(DetoxUtils) dtx_demangledCallStackSymbols] + 37
	1   Detox                               0x0000000108fa4e50 __DTXHandleCrash + 464
	2   Detox                               0x0000000108fa5595 __DTXHandleSignal + 59
	3   libsystem_platform.dylib            0x00007fff60335d7d _sigtramp + 29
	4   libsystem_c.dylib                   0x00007fff8000a728 __global_locale + 0
	5   CoreFoundation                      0x00007fff20492833 -[__NSDictionaryM dealloc] + 128
	6   libobjc.A.dylib                     0x00007fff20191604 objc_object::sidetable_release(bool, bool) + 174
	7   libobjc.A.dylib                     0x00007fff2018d8ad _object_remove_assocations + 562
	8   libobjc.A.dylib                     0x00007fff2018ac02 objc_destructInstance + 84
	9   libobjc.A.dylib                     0x00007fff20190d88 -[NSObject dealloc] + 21
	10  Foundation                          0x00007fff207f7ca1 -[NSOperation dealloc] + 281
	11  libobjc.A.dylib                     0x00007fff20191604 objc_object::sidetable_release(bool, bool) + 174
	12  libsystem_blocks.dylib              0x00007fff20080432 _Block_release + 128
	13  libsystem_blocks.dylib              0x00007fff20080432 _Block_release + 128
	14  libsystem_blocks.dylib              0x00007fff20080432 _Block_release + 128
	15  MyhealthiApp                        0x0000000108365b60 __destroy_helper_block_e8_32s40s48s + 24
	16  libsystem_blocks.dylib              0x00007fff20080432 _Block_release + 128
	17  libsystem_blocks.dylib              0x00007fff20080432 _Block_release + 128
	18  libdispatch.dylib                   0x00007fff201078df _dispatch_client_callout + 8
	19  libdispatch.dylib                   0x00007fff2010df5c _dispatch_lane_serial_drain + 1042
	20  libdispatch.dylib                   0x00007fff2010e98c _dispatch_lane_invoke + 400
	21  libdispatch.dylib                   0x00007fff20118f81 _dispatch_workloop_worker_thread + 772
	22  libsystem_pthread.dylib             0x00007fff6034045d _pthread_wqthread + 314
	23  libsystem_pthread.dylib             0x00007fff6033f42f start_wqthread + 15
)
detox[52662] ERROR: [Client.js/ERROR] Detox can't seem to connect to the test app(s)!
```

- detox: 18.20.2
- React Native: 0.63.4
- react-native-firebase 11.5.0
- Node: 14.16.0
- Device: simluator iPhone 12 Pro
- Xcode: 12.5.1
- iOS: 14.5
- macOS: 11.5.2
- Mac chip: intel i7

**Comment by sergiomap** [comment]
@alon-ha any update about this?

**Comment by alon-ha** [comment]
Not yet, I tried to debug the provided example @ball-hayden provided few times before, but it worked for me. 
I must fix some issues on iOS15 at the following week so I will get back to this when I have some time.

**Comment by ball-hayden** [comment]
This is still very much an issue.

There's been some further discussion over on https://github.com/wix/Detox/issues/3000 and https://github.com/wix/Detox/issues/2802. I'm not sure if these are duplicates, potentially?

**Comment by param-finder** [comment]
I'm facing the same issue. I can't run detox for ios due to this. I need to find an alternative to detox now.

**Comment by sergiigorbatiuk-usc** [comment]
I also faced this issue - would be nice to have some workaround or fix for this

**Comment by joannat28** [comment]
Any updates on the issue?

**Comment by param-finder** [comment]
I guess, detox is immature to use as e2e testing, will have to go for regular Appium path. I've wasted enough time on it and it's hopeless.

**Comment by asafkorem** [comment]
**As for the Signal 11 crashes when running Detox tests with Firebase:**
We are investigating this issue doing our best efforts to solve it.
Additionally, I will try to provide a workaround very soon.

**Comment by param-finder** [comment]
I have tried removing all traces of firebase and yet its the same.

**Comment by asafkorem** [comment]
@param-finder 
Can you provide here your crash log?

**Comment by param-finder** [comment]
@asafkorem 
```
 FAIL  e2e/firstTest.e2e.js (9.291 s)
  Example
    ✕ should show hello screen after tap (1 ms)

  ● Example › should show hello screen after tap

    DetoxRuntimeError: The pending request #-1000 ("isReady") has been rejected due to the following error:

    The app has crashed, see the details below:

    Signal 11 was raised
    (
    	0   Detox                               0x00000001046e46c5 +[NSThread(DetoxUtils) dtx_demangledCallStackSymbols] + 37
    	1   Detox                               0x00000001046e7250 __DTXHandleCrash + 464
    	2   Detox                               0x00000001046e7991 __DTXHandleSignal + 59
    	3   libsystem_platform.dylib            0x00007fff6bfe7e2d _sigtramp + 29
    	4   ???                                 0x0000000000000000 0x0 + 0
    	5   DetoxSync                           0x00000001349b0ca8 +[DTXRunLoopSyncResource _existingSyncResourceWithRunLoop:clear:] + 120
    	6   DetoxSync                           0x00000001349ab734 +[DTXSyncManager _untrackCFRunLoop:] + 45
    	7   DetoxSync                           0x00000001349ab6d5 +[DTXSyncManager untrackCFRunLoop:] + 75
    	8   DetoxSync                           0x00000001349a7a0b swz_runRunLoopThread + 116
    	9   Foundation                          0x000000010558a142 __NSThread__start__ + 999
    	10  libsystem_pthread.dylib             0x00007fff6bff5514 _pthread_start + 125
    	11  libsystem_pthread.dylib             0x00007fff6bff102f thread_start + 15
    )

      1 | describe('Example', () => {
      2 |   beforeAll(async () => {
    > 3 |     await device.launchApp();
        |                  ^
      4 |   });
      5 |
      6 |   beforeEach(async () => {

      at _callee$ (firstTest.e2e.js:3:18)
      at tryCatch (../node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (../node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (../node_modules/regenerator-runtime/runtime.js:119:21)
      at tryCatch (../node_modules/regenerator-runtime/runtime.js:63:40)
      at invoke (../node_modules/regenerator-runtime/runtime.js:155:20)
      at ../node_modules/regenerator-runtime/runtime.js:190:11

detox[24907] ERROR: [cli.js] Command failed: jest --config e2e/config.json --testNamePattern '^((?!:android:).)*$' e2e

error Command failed with exit code 1.
```

**Comment by asafkorem** [comment]
@param-finder this looks like a different crash those we already know.

1. Which version of Detox are you using? Can you check your project with the latest?
2. Do you have this error when running parallel workers or always?

**Comment by param-finder** [comment]
@asafkorem 
`"detox": "^19.0.0"`
`node -v: 14.18.1`
`npm -v: 6.14.15`
MacOS Monterey 12.0.1
I have this error always.
I have used detox in a new generated react native project.

**Comment by asafkorem** [comment]
@param-finder 
Is that a new generated project? can you share this repo with me and I'll take a look at it? it will be very useful in reproducing this issue in my end. You can also help me by reproducing the flow that caused the error in our [template app](https://github.com/wix-incubator/DetoxTemplate).

Also, it would be very helpful if you'll open a new issue with the required information and this crash log ([here](https://github.com/wix/Detox/issues/new?assignees=&labels=type%3A+triage%2Fbug%2C+z_platform%3A+ios&template=bug-report---ios.md&title=)). I will investigate this issue.

**Comment by param-finder** [comment]
This is a new generated project. I've upgraded detox to 19.3.0 also and tried running it but same  result. The repo is private to my org, so I wouldn't be able to share it.

**Comment by asafkorem** [comment]
> This is a new generated project. I've upgraded detox to 19.3.0 also and tried running it but same result. The repo is private to my org, so I wouldn't be able to share it. Fyi, it was an expo generated project which I had ejected later, if that makes any difference? @asafkorem

@param-finder
Yes, it makes a difference and might be the reason for your error, please see the documentation regarding Expo projects: https://wix.github.io/Detox/docs/guide/expo/#expo

However, if you still want me to investigate this error, please open a designated issue with this information. I encourage you to reproduce this error in a public repo, it will help me a lot in investigating this.

**Comment by tomphilbin** [comment]
Hey @asafkorem, many thanks for looking into this issue! I have a repository that reproduces the issue here: https://github.com/tomphilbin/DetoxFirebaseTest

Running `detox build -c ios` and then `detox test -c ios` should lead to the `Signal 11` issue, although it may take a few runs. After removing the firebase perf module with `yarn remove @react-native-firebase/perf && cd ios && pod install && cd ..` and rerunning the tests, they should pass consistently.

My team has had this issue in our project for around a year and we have workarounds in our CI/CD pipelines to do a similar to the above. We keep our dependencies up-to-date and this issue has been consistent during our upgrades of detox.

Thanks again for your time looking into this!

**Comment by asafkorem** [comment]
Thanks @tomphilbin, I will look at this repo.

**Comment by asafkorem** [comment]
Regarding the Signal 11 crashes when Firebase/Performace is integrated, see my comment on another related issue: https://github.com/wix/Detox/issues/3000#issuecomment-991940966.

If your app crashes on DetoxSync without Firebase/Performance integrated, this might be still relevant if you are using another framework / component that does ISA Swizzling. Hopefully we'll solve this issue from DetoxSync's end soon.

**Comment by asafkorem** [comment]
https://github.com/wix/Detox/pull/3135 was merged, it should solve the issue of DetoxSync crashes when **Firebase/Performance** is integrated, it was tested on the example apps that reproduced this issue. 
@tomphilbin @shamilovtim @wdospinal CC

Once you have updated a version (to the latest/next version), if any of you are still having Signal 11 crashes on DetoxSync when Firebase/Performance is integrated, please open a new bug report with the required details.

As for the parallel `--workers` issue, I think that there's still missing information here, since I fail to reproduce it myself @denissb
Can any of the reporters reproduce this issue with our template app? https://github.com/wix-incubator/DetoxTemplate

@param-finder I will investigate soon the crashes with Expo. 
Can you please open a designated issue for this bug you have?

**Comment by ball-hayden** [comment]
Sorry to be the bearer of bad news, but we're still seeing Signal 11's with Detox 19.4.1:

```
Signal 11 was raised
(
	0   Detox                               0x000000011217b6a5 +[NSThread(DetoxUtils) dtx_demangledCallStackSymbols] + 37
	1   Detox                               0x000000011217e230 __DTXHandleCrash + 464
	2   Detox                               0x000000011217e971 __DTXHandleSignal + 59
	3   libsystem_platform.dylib            0x00007fff60335d7d _sigtramp + 29
	4   ???                                 0x0000000000000125 0x0 + 293
	5   DetoxSync                           0x0000000113f27b37 -[_DTXTimerTrampoline fire:] + 188
	6   DetoxSync                           0x0000000113f0fb70 _DTXCFTimerTrampoline + 74
	7   CoreFoundation                      0x00007fff2039178a __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
	8   CoreFoundation                      0x00007fff2039127c __CFRunLoopDoTimer + 924
	9   CoreFoundation                      0x00007fff2039081a __CFRunLoopDoTimers + 265
	10  CoreFoundation                      0x00007fff2038ae69 __CFRunLoopRun + 2013
	11  CoreFoundation                      0x00007fff2038a1a7 CFRunLoopRunSpecific + 567
	12  mobile                              0x000000010d6e34e2 +[RCTCxxBridge runRunLoop] + 962
	13  DetoxSync                           0x0000000113f1773a swz_runRunLoopThread + 291
	14  Foundation                          0x00007fff2085c8a9 __NSThread__start__ + 1068
	15  libsystem_pthread.dylib             0x00007fff603438fc _pthread_start + 224
	16  libsystem_pthread.dylib             0x00007fff6033f443 thread_start + 15
)
```

It looks similar to the original trace on this issue, but happy to open a new one if you'd prefer @asafkorem ?

**Comment by asafkorem** [comment]
Hi @ball-hayden , thanks for the report. Are you running parallel tests?

**Comment by ball-hayden** [comment]
We're running with `--workers 2 --keepLockFile`, yes.

**Comment by sergiomap** [comment]
we also still have the `Signal 11` error running tests in parallel.

**Comment by s-aleinik** [comment]
I've also got interesting behaviour:

```
..detox[73724] ERROR: [WS_ERROR] The app has crashed, see the details below:

Signal 11 was raised
(
        0   Detox                               0x000000010a4556a5 +[NSThread(DetoxUtils) dtx_demangledCallStackSymbols] + 37
        1   Detox                               0x000000010a458230 __DTXHandleCrash + 464
        2   Detox                               0x000000010a458971 __DTXHandleSignal + 59
        3   libsystem_platform.dylib            0x00007fff6da24e2d _sigtramp + 29
        4   ???                                 0x0000600001050000 0x0 + 105553133371392
        5   CoreFoundation                      0x00007fff2046fd0b -[__NSDictionaryM dealloc] + 128
        6   libobjc.A.dylib                     0x00007fff2019f9f7 objc_object::sidetable_release(bool, bool) + 177
        7   libobjc.A.dylib                     0x00007fff2019bcb3 _object_remove_assocations + 578
        8   libobjc.A.dylib                     0x00007fff20199207 objc_destructInstance + 84
        9   libobjc.A.dylib                     0x00007fff2019f187 -[NSObject dealloc] + 21
        10  libobjc.A.dylib                     0x00007fff2019f9f7 objc_object::sidetable_release(bool, bool) + 177
        11  CFNetwork                           0x00007fff23e28704 _CFHTTPMessageSetResponseProxyURL + 12873
        12  libsystem_blocks.dylib              0x00007fff2008b434 _Block_release + 130
        13  Foundation                          0x00007fff207e2180 -[NSBlockOperation dealloc] + 64
        14  libobjc.A.dylib                     0x00007fff2019f9f7 objc_object::sidetable_release(bool, bool) + 177
        15  libobjc.A.dylib                     0x00007fff201a1175 AutoreleasePoolPage::releaseUntil(objc_object**) + 175
        16  libobjc.A.dylib                     0x00007fff201a1064 objc_autoreleasePoolPop + 185
        17  CFNetwork                           0x00007fff24060e96 __CFTubeSetTubeTypeNotifier + 106626
        18  CFNetwork                           0x00007fff24060d95 __CFTubeSetTubeTypeNotifier + 106369
        19  CFNetwork                           0x00007fff23f8bfd5 _CFNetworkSetATSContext + 52017
        20  CFNetwork                           0x00007fff23f8e36a _CFNetworkSetATSContext + 61126
        21  CFNetwork                           0x00007fff23f8e2da _CFNetworkSetATSContext + 60982
        22  CFNetwork                           0x00007fff23f886a5 _CFNetworkSetATSContext + 37377
        23  DetoxSync                           0x000000010a253b27 ____detox_sync_dispatch_wrapper_block_invoke + 23
        24  libdispatch.dylib                   0x00007fff2011265a _dispatch_call_block_and_release + 12
        25  libdispatch.dylib                   0x00007fff2011383a _dispatch_client_callout + 8
        26  libdispatch.dylib                   0x00007fff20119e0c _dispatch_lane_serial_drain + 743
        27  libdispatch.dylib                   0x00007fff2011aa07 _dispatch_lane_invoke + 455
        28  libdispatch.dylib                   0x00007fff2012557d _dispatch_workloop_worker_thread + 772
        29  libsystem_pthread.dylib             0x00007fff6da2f074 _pthread_wqthread + 326
        30  libsystem_pthread.dylib             0x00007fff6da2dffb start_wqthread + 15
)
```

There is **Signal 11** error. But all my tests are running successfully after it.

RN: 0.66.2
Detox 19.4.2

Firebase packages:

@react-native-firebase/analytics: 12.0.0
@react-native-firebase/app: 12.0.0
@react-native-firebase/crashlytics: 12.0.0
@react-native-firebase/dynamic-links: 12.0.0
@react-native-firebase/messaging: 12.0.0
@react-native-firebase/perf: 12.0.0
@react-native-firebase/remote-config: 12.0.0

**Comment by s-aleinik** [comment]
I've also noticed interesting behaviour:

```
import { BeforeAll, Before, AfterAll } from '@cucumber/cucumber';

BeforeAll({ timeout: 60 * 1000 }, async () => {
  await init();
});
```

When timeout is 60 seconds - crash seems to be appearing on init() function. But if timeout is 120 seconds - crash happens during the tests

**Comment by s-aleinik** [comment]
After updating all the above `@react-native-firebase` packages to `14.2.4` issue was resolved for me

**Comment by ball-hayden** [comment]
We're still seeing Signal 11s intermittently.

**Comment by ball-hayden** [comment]
Still seeing this.
I still haven't been able to narrow it down though.

**Comment by ball-hayden** [comment]
Yeah - @d4vidi there's some more information in #3207.

Both myself and @shamilovtim are struggling to isolate what causes this, so I appreciate there isn't a huge amount that can be done at the moment.

It is still an issue, however, so IMHO shouldn't be marked as stale.

**Comment by d4vidi** [comment]
Closing in favor of #3207
